pkg/settings/setting.go

```diff
@@ -197,17 +197,54 @@ type Setting interface {
 	EncodedDefault() string
 
 	Description() string
+
+	Visibility() Visibility
+
 	setDescription(desc string)
 	setSlotIdx(slotIdx int)
 	getSlotIdx() int
-	Hidden() bool
+
+	// IsReportable indicates whether the setting can be *listed*
```

Comment:
I realize you just renamed this and didn't introduce it here, but since we we are reworking/documenting the API here: I think we should re-define the behavior of this flag to be that _the value_ of this setting defaults to being hidden / redacted in listings / screenshots / telemetry, not that the setting itself is. I think it would be valuable to an operator and telemetry to know that a certain setting exists and/or is set, even if we think we should default to hiding the value from broad listings and telemetry.

From that perspective, I'm fine with the "Confidential" name, but also fine with "Reportable"

Reply:
Done.